File: src/Areas.Domain/Regions/RegionDetails.cs
using Areas.Domain.Regions.Enums;

namespace Areas.Domain.Regions;


public class RegionDetails(
    Guid id,
    String name,
    FederalDistrict federalDistrict,
    String[] plateCodes,
    Int32 localityCount
)
{
    public Guid Id { get; } = id;
    public String Name { get; } = name;
    public FederalDistrict FederalDistrict { get; } = federalDistrict;
    public String[] PlateCodes { get; } = plateCodes;
    public Int32 LocalityCount { get; } = localityCount;
}

File: src/Areas.BackOffice/client-app/src/domain/regions/regionDetails.ts
import { Page } from "../../tools/types/page";
import { FederalDistrict } from "./enums/federalDistrict";

export class RegionDetails {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly federalDistrict: FederalDistrict,
    public readonly plateCodes: string[],
    public readonly localityCount: number
  ) {}
}

export function mapToRegionDetailsPage(data: any): Page<RegionDetails> {
  return Page.convert(data, mapToRegionDetails);
}

export function mapToRegionDetailList(data: any[]): RegionDetails[] {
  return data.map(mapToRegionDetails);
}

export function mapToRegionDetails(data: any): RegionDetails {
  return new RegionDetails(
    data.id,
    data.name,
    data.federalDistrict,
    data.plateCodes,
    data.localityCount
  );
}
